src/material/progress-bar/progress-bar.scss

```diff
@@ -61,6 +61,10 @@ $mat-progress-bar-piece-animation-duration: 250ms !default;
     transform-origin: top left;
     transition: transform $mat-progress-bar-piece-animation-duration ease;
 
+    // Prevents elements from extra repainting due to `transition`.
+    will-change: transform;
+    z-index: 1;
```

Comment:
I just tried it out, but I can see the other progress bars repainting even after these changes. Would it make sense to put a `translateZ(0)` on the parent instead? Also for what it's worth, I'm sure how what performance implications this would have, considering that the element is relatively small and you usually only have one of them at a time.